flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -177,20 +177,44 @@ public boolean isCheckpointingEnabled() {
      * Gets the checkpointing mode (exactly-once vs. at-least-once).
      *
      * @return The checkpointing mode.
+     * @deprecated Use {@link #getConsistencyMode} instead.
      */
-    public CheckpointingMode getCheckpointingMode() {
+    @Deprecated
+    public org.apache.flink.streaming.api.CheckpointingMode getCheckpointingMode() {
         return configuration.get(ExecutionCheckpointingOptions.CHECKPOINTING_MODE);
     }
 
     /**
      * Sets the checkpointing mode (exactly-once vs. at-least-once).
      *
      * @param checkpointingMode The checkpointing mode.
+     * @deprecated Use {@link #setConsistencyMode} instead.
      */
-    public void setCheckpointingMode(CheckpointingMode checkpointingMode) {
+    @Deprecated
+    public void setCheckpointingMode(
+            org.apache.flink.streaming.api.CheckpointingMode checkpointingMode) {
         configuration.set(ExecutionCheckpointingOptions.CHECKPOINTING_MODE, checkpointingMode);
     }
 
+    /**
+     * Gets the checkpointing consistency mode (exactly-once vs. at-least-once).
+     *
+     * @return The checkpointing mode.
+     */
+    public CheckpointingMode getConsistencyMode() {
```

Comment:
@masteryhx  Personally, I'd be fine with name the enum `CheckpointingConsistencyMode`, but that's a breaking change and needs more discussion. My current plan is to keep the `CheckpointingMode` here, and provide API with 'Consistency' in its name just for avoiding conflicts. We will reuse the original API name later in 2.0, WDYT?

Reply:
I just have a minor concern that users may lost about `CheckpointingMode` and `ConsistencyMode`.
How about renaming this to `CheckpointingConsistencyMode` and adding some descriptions about the relationship between them ?

Reply:
I'd change the API name `get/setConsistencyMode` to `get/setCheckpointingConsistencyMode` and add some description showing the relationship between the name `CheckpointingConsistencyMode` and `CheckpointingMode`. But I'd suggest keep `CheckpointingMode` as it is since it is a first class citizen of Flink and widely accepted by users. WDYT?

Reply:
This looks good to me.